Src/WindowsServer/WindowsServer.Shared/Implementation/EnvironmentVariableMonitor.cs

```diff
@@ -0,0 +1,111 @@
+﻿namespace Microsoft.ApplicationInsights.WindowsServer.Implementation
+{
+    using System;
+    using System.Collections.Concurrent;
+    using System.Collections.Generic;
+    using System.Threading;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing;
+
+    /// <summary>
+    /// Utility to monitor the value of environment variables which may change 
+    /// during the run of an application. Checks the environment variables 
+    /// intermittently.
+    /// </summary>
+    internal abstract class EnvironmentVariableMonitor : IDisposable
+    {
+        // Environment variables tracked by this monitor.
+        protected readonly ConcurrentDictionary<string, string> CheckedValues;
+
+        // enabled flag primarily used during dispose
+        protected volatile bool isEnabled = true;
+
+        // how often we allow the code to re-check the environment
+        protected TimeSpan checkInterval;
+
+        // timer object that will periodically update the environment variables
+        private readonly Timer environmentCheckTimer;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EnvironmentVariableMonitor" /> class.
+        /// </summary>
+        public EnvironmentVariableMonitor(IEnumerable<string> envVars, TimeSpan checkInterval)
```

Comment:
Constructor of an abstract class should be protected

Reply:
Yes of course, thanks for the catch.